clients/admin-ui/src/pages/subject-request/[id].tsx

```diff
@@ -8,94 +8,67 @@ import {
 } from '@fidesui/react';
 import type { NextPage } from 'next';
 import Head from 'next/head';
-import { getSession } from 'next-auth/react';
+import { useRouter } from 'next/router';
 import React from 'react';
 
-import { useAppSelector } from '../../app/hooks';
-import { wrapper } from '../../app/store';
+import ProtectedRoute from '../../features/auth/ProtectedRoute';
 import NavBar from '../../features/common/NavBar';
-import {
-  privacyRequestApi,
-  selectPrivacyRequestFilters,
-  setRequestId,
-  setVerbose,
-  useGetAllPrivacyRequestsQuery,
-} from '../../features/privacy-requests';
+import { useGetAllPrivacyRequestsQuery } from '../../features/privacy-requests';
 import SubjectRequest from '../../features/subject-request/SubjectRequest';
-import { assignToken, setUser } from '../../features/user';
 
-const SubjectRequestDetails: NextPage<{}> = () => {
-  const filters = useAppSelector(selectPrivacyRequestFilters);
-  const { data } = useGetAllPrivacyRequestsQuery(filters);
+const useSubjectRequestDetails = () => {
+  const router = useRouter();
+  const { id = '' } = router.query;
+  const { data } = useGetAllPrivacyRequestsQuery({
+    id: Array.isArray(id) ? id[0] : id,
+    verbose: true,
+  });
```

Comment:
In my functional testing I found that this page was doing 2 requests to the backend. The first was with no filtering id so it would get back multiple requests. The second would have the id from the url as expected so it would filter and get the data.

This update makes it skip getting the data the url until the `const { id = '' } = router.query;` line above has parsed the id as expected. Now it will only do 1 request.

```suggestion
  const { data } = useGetAllPrivacyRequestsQuery(
    {
      id: Array.isArray(id) ? id[0] : id,
      verbose: true,
    },
    {
      skip: id === '',
    }
  );
```